dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,202 @@
+        if self.root == self.SPACE_APPDATA.lower():
+            default_scopes = self.SCOPE_APPDATA
+            space = self.SPACE_APPDATA
+        else:
+            default_scopes = self.SCOPE_DRIVE
+            space = self.SPACE_DRIVE
+        if Config.SECTION_GDRIVE_CREDENTIALPATH not in config:
+            shared_token_warning()
+            credentialpath = config.get(
+                Config.SECTION_GDRIVE_CREDENTIALPATH,
+                self.DEFAULT_CREDENTIALPATH,
+            )
+        scopes = config.get(Config.SECTION_GDRIVE_SCOPES, default_scopes)
+        # scopes should be a list and it is space-delimited in all
+        # configs, and `.split()` also works for a single-element list
+        scopes = scopes.split()
+
+        core_config = self.repo.config.config[Config.SECTION_CORE]
+        oauth2_flow_runner = core_config.get(
+            Config.SECTION_CORE_OAUTH2_FLOW_RUNNER, "console"
+        )
+
+        self.client = GDriveClient(
+            space,
+            config.get(Config.SECTION_GDRIVE_OAUTH_ID, self.DEFAULT_OAUTH_ID),
+            credentialpath,
+            scopes,
+            oauth2_flow_runner,
+        )
+
+    def get_file_checksum(self, path_info):
+        metadata = self.client.get_metadata(path_info, fields=["md5Checksum"])
+        return metadata["md5Checksum"]
+
+    def exists(self, path_info):
+        return self.client.exists(path_info)
+
+    def batch_exists(self, path_infos, callback):
+        results = []
+        for path_info in path_infos:
+            results.append(self.exists(path_info))
+            callback.update(str(path_info))
+        return results
+
+    def list_cache_paths(self):
+        try:
+            root = self.client.get_metadata(self.path_info)
+        except GDriveResourceNotFound as e:
+            logger.debug("list_cache_paths: {}".format(e))
+        else:
+            prefix = self.path_info.path
+            for i in self.client.list_children(root["id"]):
+                yield prefix + "/" + i
+
+    @only_once
+    def mkdir(self, parent, name):
+        return self.client.mkdir(parent, name)
+
+    def makedirs(self, path_info):
+        parent = path_info.netloc
+        parts = iter(path_info.path.split("/"))
+        current_path = ["gdrive://" + path_info.netloc]
+        for part in parts:
+            try:
+                metadata = self.client.get_metadata(
+                    self.path_cls.from_parts(
+                        self.scheme, parent, path="/" + part
+                    )
+                )
+            except GDriveResourceNotFound:
+                break
+            else:
+                current_path.append(part)
+                if not metadata_isdir(metadata):
+                    raise GDriveError(
+                        "{} is not a folder".format("/".join(current_path))
+                    )
+                parent = metadata["id"]
+        to_create = [part] + list(parts)
```

Comment:
should it be sublist of parts here?

Reply:
parts is a partially consumed iterator, but yeah, I'll rewrite it

Reply:
The exception/break condition was also valid, but DeepSource suggests too that iterating over iterator with the break/else construct and using the loop variable is something not readable and bug-risky. :)